src/quacc/calculators/espresso/espresso.py

```diff
@@ -101,11 +109,56 @@ def read_results(self, directory: Path | str) -> dict[str, Any]:
         dict
             The results dictionnary
         """
+
         results = read(Path(directory) / self.outputname, binary=self.binary)
         if "energy" not in results:
             results["energy"] = None
         return results
 
+    def _outdir_handler(self, parameters: dict[str, Any]) -> dict[str, Any]:
+        """
+        Function that handles the various outdir of espresso binaries.
+        If the user-supplied path are absolute, they are resolved and checked
+        against the scratch_dir and results_dir. If they are relative, they are
+        resolved against the scratch_dir and results_dir. If they are not in
+        either of those directories, they are ignored. This case is discouraged,
+        are encouraged to use quacc's scratch_dir and results_dir.
+
+        Parameters
+        ----------
+        parameters
+            User-supplied kwargs
+
+        Returns
+        -------
+        dict[str, Any]
+            The merged kwargs
+        """
+
+        input_data = parameters.get("input_data", {})
+
+        cwd = Path.cwd()
```

Comment:
I try to avoid relying on `cwd` whenever possible because it can be quite unexpected at times. Any reason why you can't use `Path(directory)` that was already present in `write_input`?

Reply:
No reason to use directory in this case (it will always be equal to '.' as we force it to be), a simple cwd = Path().expanduser().resolve() works, but isn't what .cwd() is doing under the hood anyway?

I suspect you might like the last commit better?

Reply:
@tomdemeyere: I had suggested `directory` just in case we eventually/somehow supported the `directory` kwarg, so that the change would be reflected elsewhere. But you're right, `directory` is forced to be `"."`, which is `cwd`, so it does not really matter in the end.

`Path().expanduser().resolve()` will do the same as `.cwd()`, as you noted. I suppose it does not matter all that much in the end since we probably won't ever support the `directory` kwarg.